octodns_edgedns/__init__.py

```diff
@@ -253,10 +277,16 @@ def _apply(self, plan):
             self._dns_client.zone_changelist_create(zone_name)
             self._dns_client.zone_changelist_submit(zone_name)
 
+        if self.use_changelist:
+            self._dns_client.zone_changelist_create(zone_name, overwrite=True)
```

Comment:
Doesn't this always recreate the zone or is it more of a create if not exists sort of thing?